MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/snapshot/SnapshotShareActivity.java

```diff
@@ -0,0 +1,184 @@
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_snapshot_share);
+
+    cameraFab = findViewById(R.id.camera_share_snapshot_image_fab);
+    cameraFab.setImageResource(R.drawable.ic_camera);
+
+    hasStartedSnapshotGeneration = false;
+
+    mapView = findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+
+    // Set a callback for when MapboxMap is ready to be used
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(final MapboxMap mapboxMap) {
+
+        SnapshotShareActivity.this.mapboxMap = mapboxMap;
+
+        // When user clicks the map, start the snapshotting process with the given parameters
+        cameraFab.setOnClickListener(new View.OnClickListener() {
+          @Override
+          public void onClick(View view) {
+
+            if (!hasStartedSnapshotGeneration) {
+              hasStartedSnapshotGeneration = true;
+              Toast.makeText(SnapshotShareActivity.this, R.string.loading_snapshot_image, Toast.LENGTH_LONG).show();
+              startSnapShot(
+                mapboxMap.getProjection().getVisibleRegion().latLngBounds,
+                mapView.getMeasuredHeight(),
+                mapView.getMeasuredWidth());
+            }
+          }
+        });
+      }
+    });
+  }
+
+  /**
+   * Creates bitmap from given parameters, and creates a notification with that bitmap
+   *
+   * @param latLngBounds of map
+   * @param height       of map
+   * @param width        of map
+   */
+  private void startSnapShot(LatLngBounds latLngBounds, int height, int width) {
+    if (mapSnapshotter == null) {
+      // Initialize snapshotter with map dimensions and given bounds
+      MapSnapshotter.Options options =
+        new MapSnapshotter.Options(width, height).withRegion(latLngBounds).withStyle(mapboxMap.getStyleUrl());
+
+      mapSnapshotter = new MapSnapshotter(SnapshotShareActivity.this, options);
+    } else {
+      // Reuse pre-existing MapSnapshotter instance
+      mapSnapshotter.setSize(width, height);
+      mapSnapshotter.setRegion(latLngBounds);
+      mapSnapshotter.setRegion(latLngBounds);
+    }
+
+    mapSnapshotter.start(new MapSnapshotter.SnapshotReadyCallback() {
+      @Override
+      public void onSnapshotReady(MapSnapshot snapshot) {
+
+        Bitmap bitmapOfMapSnapshotImage = snapshot.getBitmap();
+
+        Uri bmpUri = getLocalBitmapUri(bitmapOfMapSnapshotImage);
+
+        Intent shareIntent = new Intent();
+        shareIntent.putExtra(Intent.EXTRA_STREAM, bmpUri);
```

Comment:
Apply the following change to fix the crash while sharing on GoogleMaps and some others that were crashing):

Add: shareIntent.setAction(Intent.ACTION_SEND);
Modify: shareIntent.setType("image/png");

Reply:
This works 🎉  Thanks for looking into this